execute-sql-workitem/src/main/java/org/jbpm/process/workitem/executesql/ExecuteSqlWorkItemHandler.java

```diff
@@ -0,0 +1,163 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.executesql;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.naming.InitialContext;
+import javax.sql.DataSource;
+
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.kie.internal.runtime.Cacheable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "ExecuteSQLDefinitions.wid", name = "ExecuteSQL",
+        displayName = "ExecuteSQL",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.executesql.ExecuteSqlWorkItemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "SQLStatement", required = true),
+                @WidParameter(name = "MaxResults"),
+                @WidParameter(name = "ColumnSeparator")
+        },
+        results = {
+                @WidResult(name = "Result")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "database,fetch,sql,execute",
+                action = @WidAction(title = "Execute SQL statements")
+        ))
+public class ExecuteSqlWorkItemHandler extends AbstractLogOrThrowWorkItemHandler implements Cacheable {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExecuteSqlWorkItemHandler.class);
+    private static final String RESULT = "Result";
+    private static final int DEFAULT_MAX_RESULTS = 10;
+    private static final String DEFAULT_COLUMN_SEPARATOR = ",";
+    private ResultSet resultSet;
+    private Connection connection;
+    private PreparedStatement statement;
```

Comment:
connection, prepared statement and resultset should be created and closed within the execute method